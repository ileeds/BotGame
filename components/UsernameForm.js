import React, { Component } from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';
import { setUsername } from '../actions';
import {
	renderButton,
	renderErrorAndLoading
} from '../appUtils/renderFunctions';
import { CardSection, Input } from './common';
import { backgroundColor } from '../appUtils/puppet';

class UsernameForm extends Component {
	state = { username: '' };

	render() {
		return (
			<View style={{ backgroundColor, flex: 1 }}>
				<View style={{ flex: 0.5 }} />
				<CardSection>
					<Input
						placeholder="Username"
						onChangeText={username => this.setState({ username })}
						value={this.state.username}
					/>
				</CardSection>

				<CardSection>
					{renderButton('Submit', () => {
						this.props.setUsername(this.state.username);
					})}
				</CardSection>

				{renderErrorAndLoading(this.props.error, this.props.loading)}
				<View style={{ flex: 2 }} />
			</View>
		);
	}
}

const mapStateToProps = ({ auth }) => {
	const { error, loading } = auth;
	return { error, loading };
};

export default connect(
	mapStateToProps,
	{ setUsername }
)(UsernameForm);
