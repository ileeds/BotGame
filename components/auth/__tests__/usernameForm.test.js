import React from "react";
import { create } from "react-test-renderer";
import configureStore from "redux-mock-store";
import UsernameForm from "../UsernameForm";

jest.useFakeTimers();

const mockStore = configureStore();
const initialState = {
  auth: {
    error: "",
    loading: false
  }
};
const store = mockStore(initialState);

let component;

beforeEach(() => {
  component = create(<UsernameForm store={store} />);
});

it("renders correctly", () => {
  expect(component).toMatchSnapshot();
});
