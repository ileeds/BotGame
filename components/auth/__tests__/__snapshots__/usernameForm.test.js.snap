// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#3E5382",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.5,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#CBC9CE",
          "borderRadius": 5,
          "borderWidth": 0.1,
          "flex": 1,
          "flexDirection": "row",
          "height": 80,
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        autoFocus={true}
        onChangeText={[Function]}
        placeholder="Username"
        style={
          Object {
            "color": "black",
            "flex": 2,
            "fontSize": 36,
            "letterSpacing": 2,
            "lineHeight": 40,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        value=""
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4D608B",
          "borderRadius": 5,
          "borderWidth": 0.1,
          "flex": 1,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        Submit
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  />
</View>
`;
